hack/lib/common.sh

```diff
@@ -474,6 +474,19 @@ function kube::common::start_controller_manager {
       cloud_config_arg+=("--external-cloud-volume-plugin=${CLOUD_PROVIDER}")
       cloud_config_arg+=("--cloud-config=${CLOUD_CONFIG}")
     fi
+
+    if [[ -z "${IS_RESOURCE_PARTITION}" ]]; then
+       MASTER_ENDPOINT="https://${API_HOST}:${API_SECURE_PORT}"
```

Comment:
Perhaps MASTER_ENDPOINT only need to be set once in common.sh?

Reply:
this is refactored.